frontend/components/Infobox.vue

```diff
@@ -0,0 +1,112 @@
+<script lang="ts" setup>
+const props = defineProps({
+  progressBarExists: {
+    type: Boolean,
+    default: true,
+  },
+  duration: {
+    type: Number,
+    default: 10000,
+  },
+});
+const infoboxIsOpen = ref(true);
+
+const progress = reactive({
+  value: 100,
+});
+
+const intervalId = ref();
+
+const toggleInfobox = () => {
+  infoboxIsOpen.value = !infoboxIsOpen.value;
+  if (intervalId.value) {
+    clearInterval(intervalId.value);
+    intervalId.value = "";
+    progress.value = 100;
+  } else if (infoboxIsOpen.value) {
+    startTimer();
+  }
+};
+
+const startTimer = () => {
+  progress.value = 100;
+  intervalId.value = setInterval(() => {
+    progress.value -= 100 / (props.duration / 10);
```

Comment:
Progressbar is not moving smoothly (also remember that 0.1 cannot be saved as such because of binary floating point numbers, thus the subtracted value is not exactly equal to 0.1!)
Maybe reconsider if a progress bar is really necessary.
Here is a screenshot of the progress values that shoes the inaccuracy:
![image](https://github.com/user-attachments/assets/3ace26e9-f009-4b3b-bde4-1034b45ad2d5)

Reply:
imo you don't notice it being not completely smooth, unless you stare at it for longer, which you wouldn't do normally when you're reading the text
and we do need it, so I don't really see a better option

Reply:
When I was reviewing your PR it was everthing but smooth, however, after testing it again now, it seems to work ok. Still, the info box closes prematurely and not when the progress bar reaches its end.

Reply:
I've just checked here again and I'm happy that the progressbar finishes at the same time that the info box closes, however, to achieve this, it jumps the remaining part of the distance and is not one fluid motion :(. (Let's talk about that in person)